src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -23,33 +32,59 @@
  */
 @Log4j2
 public class ScoreCombiner {
+    public static final Float MAX_SCORE_WHEN_NO_HITS_FOUND = 0.0f;
+    // Tie-breaker to merge multiple top docs
+    private static final Comparator<ScoreDoc> SORTING_TIE_BREAKER = (o1, o2) -> {
+        int scoreComparison = Double.compare(o1.score, o2.score);
+        if (scoreComparison != 0) {
+            return scoreComparison;
+        }
 
-    private static final Float ZERO_SCORE = 0.0f;
+        int docIdComparison = Integer.compare(o1.doc, o2.doc);
+        if (docIdComparison != 0) {
+            return docIdComparison;
+        }
+
+        // When duplicate result found then both score and doc ID are equal, return 1
+        return (o1.score == o2.score && o1.doc == o2.doc) ? 1 : 0;
```

Comment:
isn't this implicit? why not just return 1?

Reply:
No, lets not directly return 1, someone who is trying to understand the code will not understand in which condition it returned 1 while debugging. The thing is here this comparator would be having duplicate result because of the sort fields values. Therefore if two documents from different shards with same docId and same sort field values then score will be same for them therefore just return 1 to act as tie breaker.

Also here I have explained this with example https://github.com/opensearch-project/neural-search/pull/802#discussion_r1655294422